vizro-ai/src/vizro_ai/dashboard/_response_models/layout.py

```diff
@@ -0,0 +1,95 @@
+"""Layout plan model."""
+
+import logging
+from typing import List
+
+import vizro.models as vm
+
+try:
+    from pydantic.v1 import BaseModel, Field, ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import BaseModel, Field, ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+def _convert_to_grid(layout_grid_template_areas, component_ids) -> List[List[int]]:
+    component_map = {component: index for index, component in enumerate(component_ids)}
+    grid = []
+
+    for row in layout_grid_template_areas:
+        grid_row = []
+        for cell in row.split():
+            if cell == ".":
+                grid_row.append(-1)
+            else:
+                try:
+                    grid_row.append(component_map[cell])
+                except KeyError:
+                    logger.warning(f"Component {cell} not found in component_ids: {component_ids}")
+                    grid_row.append(-1)
+        grid.append(grid_row)
+
+    return grid
+
+
+class LayoutPlan(BaseModel):
+    """Layout plan model, which only applies to Vizro Components(Graph, AgGrid, Card)."""
+
+    layout_description: str = Field(
+        ...,
+        description="""
+        Description of the layout of Vizro Components(Graph, AgGrid, Card).
+        Include everything that seems to relate to this layout. If layout not specified, describe layout as N/A.
```

Comment:
Why N/A, would then actually be no returned `LayoutPlan` as in it's optional in the `PagePlan`?